tests/test_deeppath.py

```diff
@@ -258,11 +258,17 @@ def test_dget_heterogenous_dicts_in_list():
     assert dget(complex_dict, "entries/data") == [
         [{"formatted": "some-error-str"}, {"values": [{"stacktrace": {"err": "why"}}]}]
     ]
-    # The assertion below fails
-    # dget assumes each dict has the same key, when it doesn't it fails as below.
-    assert dget(complex_dict, "entries/data/values") == [
-        [[{"stacktrace": {"err": "why"}}]]
-    ]
+
+    try:
+        # The assertion below fails
+        # dget assumes each dict has the same keys, when it doesn't, it fails as below.
+        assert dget(complex_dict, "entries/data/values") == [
+            [[{"stacktrace": {"err": "why"}}]]
+        ]
+    except AssertionError as exc:
+        # xfail cannot be done at assertion level.
+        # so make xfail expect ValueError to be raised
+        raise ValueError from exc
```

Comment:
I wish pytest.xfail could be mapped to a specific assertion but i don't think it can hence this hacky way. The main point of me doing this is.

I know that the first assertion should ALWAYS pass. But the second assertion should always fail until it is fixed and to ensure that the expected assertion fails.